zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -888,18 +886,16 @@ public void removeRepository(String id) throws IOException {
   }
 
   public void removeNoteInterpreterSettingBinding(String user, String noteId) throws IOException {
-    synchronized (interpreterSettings) {
```

Comment:
@Leemoonsoo , Thanks for so much reviewing. _interpreterBindings_ is a concurrentMap and hence need not require additional locking.  I believe , its true for both the changes.
The only problem that I see n the remove function is the "check and remove" .  I changed this to a "remove" operation.